tests/smoke.rs

```diff
@@ -124,6 +137,33 @@ fn check_delete() {
     })
 }
 
+#[test]
+fn check_copy_and_move() {
+    with_maildir("maildir1", |maildir1| {
+        with_maildir("maildir2", |maildir2| {
+            let id = "1463868505.38518452d49213cb409aa1db32f53184";
+
+            // check that the message is present in maildir1 but not in maildir2
+            assert!(maildir1.find(id).is_some());
+            assert!(maildir2.find(id).is_none());
+
+            // copy the message from maildir1 to maildir2
+            maildir1.copy_to(id, &maildir2).unwrap();
+
+            // check that the message is now present in both
+            assert!(maildir1.find(id).is_some());
+            assert!(maildir2.find(id).is_some());
+
+            // move the message from maildir2 to maildir1
+            maildir2.move_to(id, &maildir1).unwrap();
+
+            // check that the message is now only present in maildir1
+            assert!(maildir1.find(id).is_some());
+            assert!(maildir2.find(id).is_none());
+        })
```

Comment:
Add cases to the test for the self-copy and self-move (where source == target) scenarios too please.